src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java

```diff
@@ -259,4 +260,90 @@ public void systemEndsPatientVisit() {
 		assertNull(visitsDashboardPage.getActiveVisit());
 		dashboardPage = visitsDashboardPage.goToPatientDashboard();
 	}
+
+	@When("a user clicks on Allergy2 link from Patient dashboard page2")
+	public void loadManageAllergy2() {
+		allergyPage = (AllergyPage) dashboardPage.clickOnAllergiesWidgetLink()
+				.waitForPage();
+	}
+
+	@Then("the system loads Allergy2 board page")
+	public void loadsManageAllergy2() {
+				assertTrue(textExists("Allergies"));
+	}
+
+	@When("a user clicks Add New Allergy2 button")
+	public void addNewAllergy2() {
+		addOrEditAllergyPage = allergyPage.clickOnAddNewAllergy();
+		addOrEditAllergyPage.enterDrug(DRUG_NAME);
+		addOrEditAllergyPage.drugId();
+	}
+
+	@And("the user clicks on the Save button2")
+	public void saveAllergy2() {
+		addOrEditAllergyPage.clickOnSaveAllergy();
+	}
+
+	@Then("the system adds a new Allergy2")
+	public void systemAddsAllergy2() {
+		assertTrue(textExists("Penicillins"));
+	}
+
+	@When("the user clicks on the delete button from the patient dashboard")
+	public void deleteAllergy2() {
+		allergyPage.clickOnDeleteAllergy();
+	}
+
+	@And("the system loads Remove Allergy dashboard")
+	public void confirmDeleteAllergy2() {
+		allergyPage.clickOnConfirmDeleteAllergy();
+	}
+
+	@Then("system displays no allergy in the Allergies table")
+	public void systemRemovesAllergy2() {
+		assertTrue(textExists("Unknown"));
+	}
+	
+	@When("a user clicks on Condition2 link from Patient dashboard page")
+	public void loadManageCondition2() {
+		conditionsPage = (ConditionsPage) dashboardPage
+				.clickOnConditionsWidgetLink().waitForPage();
+	}
```

Comment:
Crosscheck the method within the file that achieves what `loadMangeCondition2()` is a accomplishing and make use of it